src/test.js

```diff
@@ -131,57 +127,73 @@ class Test {
-        await this._loadScenario();
+        if (this.scenario) {
+            await this.LoopUtilities().loadScenario(this.scenario);
+        }
         await this._setupCGMData();
         await this._setLoopMode();
         await this._setStartScreen();
+
     }
     async _setLoopMode() {
         if (this.closedLoop) {
-            var screen = await this.OpenSettingsScreen();
-            await screen.ClosedLoop();
-            await screen.Back();
+            await this.LoopUtilities().closeLoop();
         }
     }
     async _setSimulators() {
         if (this.simulators) {
             if (this.simulators.cgm) {
-                await this.addCGM();
+                await this.LoopUtilities().addCGM();
             }
             if (this.simulators.pump) {
-                await this.addUnconfiguredPump();
+                await this.LoopUtilities().addUnconfiguredPump();
             }
         }
     }
-    async advanceScenario(cycles) {
-        await device.shake();
-        await expect(match.accessible.TextLabel(this.scenario)).toExist();
-        await match.accessible.TextLabel(this.scenario).swipe('left');
-        await match.accessible.SwipeButton('Advance ⏭').tap();
-        await match.UITextField().typeText(cycles);
-        await match.accessible.Button(this.language.general.OK).tap();
-    }
+    // async advanceScenario(cycles) {
+    //     await device.shake();
+    //     await expect(match.accessible.TextLabel(this.scenario)).toExist();
+    //     await match.accessible.TextLabel(this.scenario).swipe('left');
+    //     await match.accessible.SwipeButton('Advance ⏭').tap();
+    //     await match.UITextField().typeText(cycles);
+    //     await match.accessible.Button(this.language.general.OK).tap();
+    // }
     /**
      * @summary will load the mocked therapy settings
      */
-    async addConfiguredPump() {
-        await this.addUnconfiguredPump();
-        await this.loadTherapySettings(true);
-    }
-    async addUnconfiguredPump() {
-        await this.homeScreen.HeaderSection().Devices().AddPump();
-    }
-    async addCGM() {
-        await this.homeScreen.HeaderSection().Devices().AddCGM();
-    }
-    async openPumpScreen() {
+    // async addConfiguredPump() {
+    //     await this.useCases.addConfiguredPump();
+    //     // await this.addUnconfiguredPump();
+    //     // await this.loadTherapySettings(true);
+    // }
+    // async addUnconfiguredPump() {
+    //     await this.useCases.addConfiguredPump();
+    //     // await this.homeScreen.HeaderSection().Devices().AddPump();
+    // }
+    // async addCGM() {
+    //     await this.useCases.addCGM();
+    //     //await this.homeScreen.HeaderSection().Devices().AddCGM();
+    // }
+    // async loadTherapySettings(load) {
+    //     if (load) {
+    //         await device.shake();
+    //         await match.accessible.TextLabel('Mock Therapy Settings').tap();
+    //     }
+    // }
+    LoopUtilities() {
```

Comment:
Ah, I see now. This is a function on `test` that returns the single instance of `this.loopUtilities`. Okay, that makes more sense, but it confused me. Perhaps just make it a property rather than a function?

I guess I'm just not clear about what decides whether something is a property or a function. For example, `this.homeScreen.HeaderSection().Devices().AddCGM()`. The last `AddCGM()` makes sense to me as a function since it is an action to "add cgm". The others are unclear. Why is `homeScreen` a property while `HeaderSection()` and `Devices()` are functions?

(Maybe I'm just not seeing it since I'm sleep deprived right now. 😉 )

Reply:
yeah I agree ... its probably more of an evolution and needs more cleanup for consistency. I think I have been lazy and a lot of these are actually getter functions. 
`getHeaderSection()` `getDevices()` ??